helper/token.js

```diff
@@ -0,0 +1,79 @@
+const promise = require('bluebird');
+const dbHelper = require('../helper/database');
+const request = require('request');
+const tokenCollection = 'token';
+const canvasUrl = 'http://canvas.differ.chat/api/v1/users/36/courses?access_token=';
+
+let getToken = (token) => {
+    return new promise(function (resolve, reject) {
+        dbHelper
+            .dbClient
+            .collection(tokenCollection)
+            .findOne(
+                {
+                    token: token
+                }, {}, function (err, foundToken) {
+                    if (err) {
+                        return reject(err);
+                    }
+                    return resolve(!!foundToken);
+                });
+    })
+};
+
+let saveToken = (token) => {
+    return dbHelper
+        .dbClient
+        .collection(tokenCollection)
+        .insert({
+            token: token
+        })
+        .then(() => {
+            return {
+                success: true
+            };
+        });
+};
+
+let updateTokenWithCourses = (token, courses) => {
+    return dbHelper
+        .dbClient
+        .collection(tokenCollection)
+        .findOneAndUpdate({
+            token: token
+        }, {
+            $set: {
+                courses: courses
+            }
+        }, {
+            upsert: false,
+            returnNewDocument: true
+        })
+        .then(courses => {
+            return courses.value;
+        });
+};
+
+let importCourses = (token) => {
+    return new promise(function (resolve, reject) {
+        request.get(canvasUrl, {
```

Comment:
this logic should probably be in another service. Instead of everything being in helper-files, create a folder services/ and persistence/. Put this method in services/ and the ones that deal with database-operations in persistence/. This means that the API can orchestrate the whole flow. 

in api:
CanvasService
.importCourses()
.then(() => {
   TokenStorage.updateTokenWithCourses();
})